support-workers/src/typescript/services/salesforce.ts

```diff
@@ -136,64 +145,95 @@ export class SalesforceService {
 		user: User,
 	): Promise<SuccessfulUpsertResponse> | null {
 		if (giftRecipient?.firstName && giftRecipient.lastName) {
-			const giftRecipientContact: DeliveryContactRecordRequest = {
-				AccountId: contactRecord.AccountId,
-				Email: giftRecipient.email,
-				Salutation: giftRecipient.title,
-				FirstName: giftRecipient.firstName,
-				LastName: giftRecipient.lastName,
-				MailingStreet: user.deliveryAddress
-					? getAddressLine(user.deliveryAddress)
-					: null,
-				MailingCity: user.deliveryAddress?.city ?? null,
-				MailingState: user.deliveryAddress?.state ?? null,
-				MailingPostalCode: user.deliveryAddress?.postCode ?? null,
-				MailingCountry: user.deliveryAddress
-					? getCountryNameByIsoCode(user.deliveryAddress.country)
-					: null,
-				RecordTypeId: '01220000000VB50AAG',
-			};
+			const giftRecipientContact: DeliveryContactRecordRequest =
+				createDeliveryContactRecordRequest(contactRecord, giftRecipient, user);
 			return this.upsert(giftRecipientContact);
 		}
 		return null;
 	}
 }
+const createBillingAddressFields = (user: User) => ({
+	OtherStreet: getAddressLine(user.billingAddress),
+	OtherCity: user.billingAddress.city ?? null,
+	OtherState: user.billingAddress.state ?? null,
+	OtherPostalCode: user.billingAddress.postCode ?? null,
+	OtherCountry: getCountryNameByIsoCode(user.billingAddress.country) ?? null,
+});
 
-export const createContactRecordRequest = (
+const createMailingAddressFields = (user: User) => {
+	return {
+		MailingStreet: user.deliveryAddress
+			? getAddressLine(user.deliveryAddress)
+			: undefined,
+		MailingCity: user.deliveryAddress?.city ?? null,
+		MailingState: user.deliveryAddress?.state ?? null,
+		MailingPostalCode: user.deliveryAddress?.postCode ?? null,
+		MailingCountry: user.deliveryAddress
+			? getCountryNameByIsoCode(user.deliveryAddress.country)
+			: null,
+	};
+};
+
+const getContactType = (user: User): 'Standard' | 'DeliveryOrRecipient' => {
+	if (!user.deliveryAddress) {
```

Comment:
Does this not mean that for any digital only subscription we will no longer be sending any address fields at all? I don't think that's right because we will have a country and state even for recurring contributions